src/Microsoft.Fast.Components.FluentUI/Components/FluentDialog.razor

```diff
@@ -1,9 +1,15 @@
-<fluent-dialog Modal="@Modal" @attributes="AdditionalAttributes">@ChildContent</fluent-dialog>
+<fluent-dialog Modal="@Modal" @attributes="AdditionalAttributes" hidden="@(Hidden)">@ChildContent</fluent-dialog>
 @code
 {
     [Parameter] public bool? Modal { get; set; }
 
     [Parameter] public RenderFragment ChildContent { get; set; }
 
     [Parameter(CaptureUnmatchedValues = true)] public IDictionary<string, object> AdditionalAttributes { get; set; }
+
+    public bool Hidden { get; set; } = true;
```

Comment:
The component's default hidden value is `false`, I think this should probably align unless there is a good reason not to: https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/dialog/dialog.ts#L32

Reply:
Makes sense, but it's a bit weird, do we want the component by default to be shown? 

Having it to `false` would make that the developer have to reference the dialog and on the initial rendering call `Hide()` in order to have it hidden at the start. Making it `true` ensures that the component itself it's always hidden in the start until `Show()` is called.

Is there any reason why the original fast component default value is `false`?

Reply:
I believe this is because the native `dialog` element follows the same pattern and has a `hidden` attribute that is also `false` by default. The idea being that if you add a dialog to the page, it should display itself, unless otherwise configured. So, you would typically use this in a few ways:

* Bind the hidden property using any framework and control it that way.
* Use your frameworks conditional rendering mechanism to dynamically add/remove the dialog per display needs
* Use DOM APIs to manually create and append/remove a dialog per display needs

One of our goals with the web components is to stay as close to the native platform as possible, so I don't think we can change the default value of `hidden`. I also don't want to create confusion by having the Blazor wrappers have different defaults than the underlying web components. That also creates a maintenance and documentation burden for us that we can't take on.

So, I think we do need to switch the wrapper here to match the standard component behavior. I believe that the way this is setup, Blazor devs wouldn't need to reference the element and call show/hide. They should be able to just set the `Hidden` property to false themselves and even control that through binding.

Reply:
Okay, so in order to keep consistency, you would delete the existing `Hide` and `Show` methods and set the hidden as a optional parameter of the wrapper? So the parent component handles if it's shown or not.

```csharp
[Parameter] public bool Hidden { get; set; } = false;
```

I like your approach and it makes sense, I still think that eventualy developers will have to duplicate code for each dialog they want to use, like having multiples `boolean` objects to handle what's being shown or not. In any case, I think that keeping consistency with the main framework is the best idea.

How do you want the apporach to be? As I mention before, making the `Hidden` property as an optional parameter would be the best for it.

Reply:
I think we are agreed. We can make the parameter optional with the default being the same as the underlying web component *and* keep the hide/show methods, since they mirror the hide/show on the base component. It just turns out that it's easier to mirror those in Blazor than to do the JS interop to call the originals (which do the same thing in the end).

Reply:
Updated the code with the new approach and added both cases in the test page.